src/pke/include/cryptocontext.h

```diff
@@ -2100,6 +2101,10 @@ class CryptoContextImpl : public Serializable {
    */
     Ciphertext<Element> EvalRotate(ConstCiphertext<Element> ciphertext, int32_t index) const {
         CheckCiphertext(ciphertext);
+        // rotation can be performed on 2-element ciphertexts only
+        if (ciphertext->NumberCiphertextElements() != 2) {
```

Comment:
Add it at the EvalAutomorphism level instead. This will capture EvalRotate, EvalAtIndex, and other other operations derived from EvalAutomorphism

Reply:
Addressed. the changes have been committed